e2e/__tests__/selectProjects.test.ts

```diff
@@ -0,0 +1,168 @@
+  const dir = resolve(__dirname, '..', 'select-projects');
+
+  describe('when Jest is started with `--selectProjects first-project`', () => {
+    const result = runWithJson('select-projects', [
+      '--selectProjects',
+      'first-project',
+    ]);
+    it('runs the tests in the first project only', () => {
+      expect(result.json.success).toBe(true);
+      expect(result.json.numTotalTests).toBe(1);
+      expect(result.json.testResults.map(({name}) => name)).toEqual([
+        resolve(dir, '__tests__/first-project.test.js'),
+      ]);
+    });
+    it('prints that only first-project will run', () => {
+      expect(result.stderr).toMatch(/^Running one project: first-project/);
+    });
+  });
+
+  describe('when Jest is started with `--selectProjects second-project`', () => {
+    const result = runWithJson('select-projects', [
+      '--selectProjects',
+      'second-project',
+    ]);
+    it('runs the tests in the second project only', () => {
+      expect(result.json.success).toBe(true);
+      expect(result.json.numTotalTests).toBe(1);
+      expect(result.json.testResults.map(({name}) => name)).toEqual([
+        resolve(dir, '__tests__/second-project.test.js'),
+      ]);
+    });
+    it('prints that only second-project will run', () => {
+      expect(result.stderr).toMatch(/^Running one project: second-project/);
+    });
+  });
+
+  describe('when Jest is started with `--selectProjects first-project second-project`', () => {
+    const result = runWithJson('select-projects', [
+      '--selectProjects',
+      'first-project',
+      'second-project',
+    ]);
+    it('runs the tests in the first and second projects', () => {
+      expect(result.json.success).toBe(true);
+      expect(result.json.numTotalTests).toBe(2);
+      expect(result.json.testResults.map(({name}) => name).sort()).toEqual([
+        resolve(dir, '__tests__/first-project.test.js'),
+        resolve(dir, '__tests__/second-project.test.js'),
+      ]);
+    });
+    it('prints that both first-project and second-project will run', () => {
+      expect(result.stderr).toMatch(
+        /^Running 2 projects:\n- first-project\n- second-project/,
+      );
+    });
+  });
+
+  describe('when Jest is started without providing `--selectProjects`', () => {
+    const result = runWithJson('select-projects', []);
+    it('runs the tests in the first and second projects', () => {
+      expect(result.json.success).toBe(true);
+      expect(result.json.numTotalTests).toBe(2);
+      expect(result.json.testResults.map(({name}) => name).sort()).toEqual([
+        resolve(dir, '__tests__/first-project.test.js'),
+        resolve(dir, '__tests__/second-project.test.js'),
+      ]);
+    });
+    it('does not print which projects are run', () => {
+      expect(result.stderr).not.toMatch(/^Running/);
+    });
+  });
+
+  describe('when Jest is started with `--selectProjects third-project`', () => {
+    const result = runWithJson('select-projects', [
```

Comment:
can you stick this inside the test? `describe` runs before tests start. can combine the 2 `it`s, I guess, or stick the `runWithJson` inside a `beforeAll`